crash on elastic_ecs transmission get_pagesize return None

**Describe the bug**
Against one Elasticsearch instance, when running Kestrel v1.5.10 with stix-shifter v4.6.1, a deterministic crash will occur every time a huntflow runs:

```
[ERROR] DataSourceError: STIX-shifter transmission.results() failed with message: elastic_ecs connector error => '<' not supported between instances of 'NoneType' and 'int'
please check data source config or test the query manually.
```

**To Reproduce**
Steps to reproduce the behavior:
1. stix-shifter v4.6.1
2. any pattern such as: `[process:name ='powershell.exe']`
3. default stix-shifter config
4. error happens when doing `transmission.results()`

**Expected behavior**
No error when return

**Full crashed stack**
```
23:06:21 DEBUG stix_shifter_utils.utils.error_response   File "/usr/local/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/xxx/.local/lib/python3.9/site-packages/kestrel_jupyter_kernel/__main__.py", line 5, in <module>
    IPKernelApp.launch_instance(kernel_class=KestrelKernel)
  File "/usr/local/lib/python3.9/site-packages/traitlets/config/application.py", line 664, in launch_instance
    app.start()
  File "/usr/local/lib/python3.9/site-packages/ipykernel/kernelapp.py", line 612, in start
    self.io_loop.start()
  File "/usr/local/lib/python3.9/site-packages/tornado/platform/asyncio.py", line 199, in start
    self.asyncio_loop.run_forever()
  File "/home/xxx/.local/lib/python3.9/site-packages/nest_asyncio.py", line 73, in run_forever
    self._run_once()
  File "/home/xxx/.local/lib/python3.9/site-packages/nest_asyncio.py", line 120, in _run_once
    handle._run()
  File "/usr/local/lib/python3.9/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/local/lib/python3.9/site-packages/tornado/ioloop.py", line 688, in <lambda>
    lambda f: self._run_callback(functools.partial(callback, future))
  File "/usr/local/lib/python3.9/site-packages/tornado/ioloop.py", line 741, in _run_callback
    ret = callback()
  File "/usr/local/lib/python3.9/site-packages/tornado/gen.py", line 814, in inner
    self.ctx_run(self.run)
  File "/usr/local/lib/python3.9/site-packages/tornado/gen.py", line 775, in run
    yielded = self.gen.send(value)
  File "/usr/local/lib/python3.9/site-packages/ipykernel/kernelbase.py", line 365, in process_one
    yield gen.maybe_future(dispatch(*args))
  File "/usr/local/lib/python3.9/site-packages/tornado/gen.py", line 234, in wrapper
    yielded = ctx_run(next, result)
  File "/usr/local/lib/python3.9/site-packages/ipykernel/kernelbase.py", line 268, in dispatch_shell
    yield gen.maybe_future(handler(stream, idents, msg))
  File "/usr/local/lib/python3.9/site-packages/tornado/gen.py", line 234, in wrapper
    yielded = ctx_run(next, result)
  File "/usr/local/lib/python3.9/site-packages/ipykernel/kernelbase.py", line 543, in execute_request
    self.do_execute(
  File "/home/xxx/.local/lib/python3.9/site-packages/kestrel_jupyter_kernel/kernel.py", line 47, in do_execute
    outputs = self.kestrel_session.execute(code)
  File "/home/xxx/.local/lib/python3.9/site-packages/kestrel/session.py", line 268, in execute
    return self._execute_ast(ast)
  File "/home/xxx/.local/lib/python3.9/site-packages/kestrel/session.py", line 406, in _execute_ast
    output_var_struct, display = execute_cmd(stmt, self)
  File "/home/xxx/.local/lib/python3.9/site-packages/kestrel/codegen/commands.py", line 101, in wrapper
    return func(stmt, session)
  File "/home/xxx/.local/lib/python3.9/site-packages/kestrel/codegen/commands.py", line 72, in wrapper
    ret = func(stmt, session)
  File "/home/xxx/.local/lib/python3.9/site-packages/kestrel/codegen/commands.py", line 268, in get
    rs = session.data_source_manager.query(
  File "/home/xxx/.local/lib/python3.9/site-packages/kestrel/datasource/manager.py", line 33, in query
    rs = i.query(uri, pattern, session_id, c, store)
  File "/home/xxx/.local/lib/python3.9/site-packages/kestrel_datasource_stixshifter/interface.py", line 250, in query
    result_batch = transmission.results(
  File "/home/xxx/.local/lib/python3.9/site-packages/stix_shifter/stix_transmission/stix_transmission.py", line 64, in results
    return self.entry_point.create_results_connection(search_id, offset, length)
  File "/home/xxx/.local/lib/python3.9/site-packages/stix_shifter_utils/utils/base_entry_point.py", line 79, in wrapper_func
    return func(self, *args, **kwargs)
  File "/home/xxx/.local/lib/python3.9/site-packages/stix_shifter_utils/utils/base_entry_point.py", line 268, in create_results_connection
    return self.__results_connector.create_results_connection(search_id, offset, length)
  File "/home/xxx/.local/lib/python3.9/site-packages/stix_shifter_modules/elastic_ecs/stix_transmission/connector.py", line 112, in create_results_connection
    raise e
  File "/home/xxx/.local/lib/python3.9/site-packages/stix_shifter_modules/elastic_ecs/stix_transmission/connector.py", line 92, in create_results_connection
    response = self.api_client.search_pagination(query, metadata, min(int(length), self.max_result_window))

  <class 'TypeError'> '<' not supported between instances of 'NoneType' and 'int'
```

**Direct cause**
https://github.com/opencybersecurityalliance/stix-shifter/blob/develop_4.x/stix_shifter_modules/elastic_ecs/stix_transmission/connector.py#L92
```
response = self.api_client.search_pagination(query, metadata, min(int(length), self.max_result_window))
```

Basically `self.max_result_window` is `None`. So I patched the `__init__` function from:
```
        try:
            self.max_result_window = self.get_pagesize()
        except Exception as e:
            pass
```
to
```
        try:
            ps = self.get_pagesize()
            if ps:
                self.max_result_window = ps
        except Exception as e:
            pass
```
And it is working again.

**Further debug needed**
This means for some reason, `get_pagesize()` returns `None`, which should not happen. I know the specific index queried in the Elasticsearch has `max_result_window` set to 20000.

However, when I do:
```
GET /target_index/_settings?include_defaults=true
```
I see there are 9 `max_result_window` keyword in the returned JSON and only the first one is set to 20000, the remaining of which are just 10000.

Need to check the logic of getting pagesize to finally fix it.

--- a/stix_shifter_modules/elastic_ecs/stix_transmission/api_client.py
+++ b/stix_shifter_modules/elastic_ecs/stix_transmission/api_client.py
@@ -92,10 +92,8 @@ class APIClient():
         return self.client.call_api(endpoint, 'GET', headers, data=json.dumps(data), timeout=self.timeout)
 
     def get_max_result_window(self):
-        # GET winlogbeat-*/_settings?include_defaults=true
-        endpoint = self.setting_endpoint
-        endpoint = "{}?include_defaults=true".format(endpoint)
-        return self.client.call_api(endpoint, 'GET', timeout=self.timeout)
+        max_result_window_url = self.setting_endpoint + "/index.max_result_window?include_defaults=true"
+        return self.client.call_api(max_result_window_url, 'GET', timeout=self.timeout)
 
     def set_pit(self):
         headers = dict()

--- a/stix_shifter_modules/elastic_ecs/stix_transmission/connector.py
+++ b/stix_shifter_modules/elastic_ecs/stix_transmission/connector.py
@@ -16,10 +16,7 @@ class Connector(BaseSyncConnector):
         self.api_client = APIClient(connection, configuration)
         self.logger = logger.set_logger(__name__)
         self.connector = __name__.split('.')[1]
-        try:
-            self.max_result_window = self.get_pagesize(DEFAULT_MAX_RESULTS_WINDOW_SIZE)
-        except Exception as e:
-            pass
+        self.max_result_window = self.get_pagesize(DEFAULT_MAX_RESULTS_WINDOW_SIZE)
 
     def _handle_errors(self, response, return_obj):
         response_code = response.code
@@ -53,11 +50,15 @@ class Connector(BaseSyncConnector):
                 raise e
 
     def get_pagesize(self, default_window_size):
-        response_txt = None
         return_obj = dict()
+
         try:
             response = self.api_client.get_max_result_window()
             return_obj = self._handle_errors(response, return_obj)
+        except Exception:
+            max_result_window = default_window_size
+            pass
+        else:
             if (return_obj['success']):
                 response_json = json.loads(return_obj["data"])
                 max_result_windows = []
@@ -79,13 +80,6 @@ class Connector(BaseSyncConnector):
 
             else: # land here if API call failed, e.g., no priviledge
                 max_result_window = default_window_size
-
-        except Exception as e:
-            if response_txt is not None:
-                ErrorResponder.fill_error(return_obj, message='unexpected exception', connector=self.connector)
-                self.logger.error('can not parse response: ' + str(response_txt))
-            else:
-                raise e
 
         return max_result_window
 

--- a/stix_shifter_modules/elastic_ecs/stix_transmission/connector.py
+++ b/stix_shifter_modules/elastic_ecs/stix_transmission/connector.py
@@ -57,7 +57,6 @@ class Connector(BaseSyncConnector):
             return_obj = self._handle_errors(response, return_obj)
         except Exception:
             max_result_window = default_window_size
-            pass
         else:
             if (return_obj['success']):
                 response_json = json.loads(return_obj["data"])

--- a/stix_shifter_modules/elastic_ecs/stix_transmission/connector.py
+++ b/stix_shifter_modules/elastic_ecs/stix_transmission/connector.py
@@ -4,6 +4,8 @@ import json
 from stix_shifter_utils.utils.error_response import ErrorResponder
 from stix_shifter_utils.utils import logger
 
+DEFAULT_MAX_RESULTS_WINDOW_SIZE = 10000
+
 
 class UnexpectedResponseException(Exception):
     pass
@@ -14,10 +16,8 @@ class Connector(BaseSyncConnector):
         self.api_client = APIClient(connection, configuration)
         self.logger = logger.set_logger(__name__)
         self.connector = __name__.split('.')[1]
-        self.max_result_window = 10000
-        # extract the max_result_window from elasticsearch
         try:
-            self.max_result_window = self.get_pagesize()
+            self.max_result_window = self.get_pagesize(DEFAULT_MAX_RESULTS_WINDOW_SIZE)
         except Exception as e:
             pass
 
@@ -52,7 +52,7 @@ class Connector(BaseSyncConnector):
             else:
                 raise e
 
-    def get_pagesize(self):
+    def get_pagesize(self, default_window_size):
         response_txt = None
         return_obj = dict()
         try:
@@ -60,12 +60,12 @@ class Connector(BaseSyncConnector):
             return_obj = self._handle_errors(response, return_obj)
             if (return_obj['success']):
                 response_json = json.loads(return_obj["data"])
-                max_result_windows = list()
-                if not (response_json is None):
+                max_result_windows = [default_window_size] # at least one item for [0] in max_result_window calculation
+                if response_json:
                     for index, item_json in response_json.items():
-                        if 'max_result_window' in item_json['settings']['index']:
+                        if 'index' in item_json['settings'] and 'max_result_window' in item_json['settings']['index']:
                             max_res_win = item_json['settings']['index']['max_result_window']
-                        elif 'max_result_window' in item_json['defaults']['index']:
+                        elif 'index' in item_json['defaults'] and 'max_result_window' in item_json['defaults']['index']:
                             max_res_win = item_json['defaults']['index']['max_result_window']
                         else:
                             ErrorResponder.fill_error(item_json,
@@ -75,6 +75,8 @@ class Connector(BaseSyncConnector):
                         max_result_windows.append(int(max_res_win))
                 max_result_window = sorted(max_result_windows)[0] #return the smallest max_return_window in indices
                 return max_result_window
+            else: # land here if API call failed, e.g., no priviledge
+                return default_window_size
         except Exception as e:
             if response_txt is not None:
                 ErrorResponder.fill_error(return_obj, message='unexpected exception', connector=self.connector)

--- a/stix_shifter_modules/elastic_ecs/stix_transmission/connector.py
+++ b/stix_shifter_modules/elastic_ecs/stix_transmission/connector.py
@@ -60,7 +60,7 @@ class Connector(BaseSyncConnector):
             return_obj = self._handle_errors(response, return_obj)
             if (return_obj['success']):
                 response_json = json.loads(return_obj["data"])
-                max_result_windows = [default_window_size] # at least one item for [0] in max_result_window calculation
+                max_result_windows = []
                 if response_json:
                     for index, item_json in response_json.items():
                         if 'index' in item_json['settings'] and 'max_result_window' in item_json['settings']['index']:
@@ -73,16 +73,21 @@ class Connector(BaseSyncConnector):
                                                       connector=self.connector)
                             self.logger.error('max_result_window is not set in index: ' + str(index))
                         max_result_windows.append(int(max_res_win))
-                max_result_window = sorted(max_result_windows)[0] #return the smallest max_return_window in indices
-                return max_result_window
+
+                # return the smallest max_return_window in indices
+                max_result_window = sorted(max_result_windows)[0] if max_result_windows else default_window_size
+
             else: # land here if API call failed, e.g., no priviledge
-                return default_window_size
+                max_result_window = default_window_size
+
         except Exception as e:
             if response_txt is not None:
                 ErrorResponder.fill_error(return_obj, message='unexpected exception', connector=self.connector)
                 self.logger.error('can not parse response: ' + str(response_txt))
             else:
                 raise e
+
+        return max_result_window
 
     def create_results_connection(self, query, offset, length, metadata=None):
         response_txt = None

--- a/stix_shifter_modules/elastic_ecs/tests/stix_transmission/test_elastic_ecs.py
+++ b/stix_shifter_modules/elastic_ecs/tests/stix_transmission/test_elastic_ecs.py
@@ -322,3 +322,24 @@ class TestElasticEcsConnection(unittest.TestCase, object):
 
         assert transmission.entry_point._BaseEntryPoint__results_connector.max_result_window == 20000
 
+    @patch('stix_shifter_modules.elastic_ecs.stix_transmission.api_client.APIClient.get_max_result_window')
+    def test_pagesize_exception(self, mock_result_response, mock_api_client):
+        mock_api_client.return_value = None
+        mocked_return_value = '["mock", "placeholder"]'
+        mock_result_response.return_value = ElasticEcsMockResponse(200, mocked_return_value)
+        mock_result_response.side_effect = Exception('exception')
+        config = {
+            "auth": {
+                "username": "bla",
+                "password": "bla"
+            }
+        }
+        connection = {
+            "host": "hostbla",
+            "port": 8080,
+            "selfSignedCert": "cert",
+            "indices": "index1"
+        }
+
+        transmission = stix_transmission.StixTransmission('elastic_ecs', connection, config)
+        assert transmission.entry_point._BaseEntryPoint__results_connector.max_result_window == 10000

